pkg/controllers/jivavolume_controller.go

```diff
@@ -152,8 +153,20 @@ func (r *JivaVolumeReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 					instance.Name, err.Error())
 			}
 		}
+		if err := r.moveReplicasForMissingNodes(instance); err != nil {
+			r.Recorder.Eventf(instance, corev1.EventTypeWarning,
+				"ReplicaMovement", "failed to move replica, due to error: %v", err)
+			return reconcile.Result{}, fmt.Errorf("failed to move replica %s: %s",
+				instance.Name, err.Error())
+		}
 		return reconcile.Result{}, r.getAndUpdateVolumeStatus(instance)
 	case openebsiov1alpha1.JivaVolumePhaseSyncing, openebsiov1alpha1.JivaVolumePhaseUnkown:
+		if err := r.moveReplicasForMissingNodes(instance); err != nil {
+			r.Recorder.Eventf(instance, corev1.EventTypeWarning,
+				"ReplicaMovement", "failed to move replica, due to error: %v", err)
+			return reconcile.Result{}, fmt.Errorf("failed to move replica %s: %s",
+				instance.Name, err.Error())
+		}
```

Comment:
Can we do this process only when `jiva.Status.Phase` is `Ready`? I am a bit concerned about the following cases:

- User configured Jiva with one storage replica and node is lost then we are silently marking the replica as active on new node without any events saying about the data(In this case we can throw events before moving replica).

(Or)

- Can we take input from the user via env(ENABLE_REPLICA_MOVEMENT) (or) arguments saying always move my replica, irrespective of volume state.

One more question: `volume.status.phase` is coming from previous reconciliation right?

Reply:
> is coming from previous reconciliation right?

Yes

Reply:
With the latest changes, we are fetching replica information and processing it accordingly so safe to move replicas...